src/Components/Components/src/Routing/Router.cs

```diff
@@ -93,11 +95,16 @@ public Task SetParametersAsync(ParameterView parameters)
                 throw new InvalidOperationException($"The {nameof(Router)} component requires a value for the parameter {nameof(NotFound)}.");
             }
 
+            if (OnNavigate != null) {
+                NavigationManager.OnNavigate = OnNavigate;
+            }
+
+           await NavigationManager.BeforeLocationChangeAsync();
```

Comment:
Wouldn't this load assemblies right away?

Reply:
Only for routes that return a list of assemblies to lazy-load in the `OnNavigate`. This addresses the scenario where a user lands on a page with lazy loaded dependencies for the first time, instead of navigating to it from another page.